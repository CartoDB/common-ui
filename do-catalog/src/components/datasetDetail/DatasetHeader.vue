<template>
  <header class="grid u-flex__justify--center u-mb--36 u-mb--20--tablet u-mt--36">
    <div class="grid-cell grid-cell--col9 grid-cell--col8--tablet">
      <nav class="breadcrumbs">
        <p class="text is-caption is-txtMainTextColor" v-if="!isGeography">
          <span class="title is-txtMainTextColor">{{
            dataset.category_name
          }}</span>
          /
          <span class="text is-txtMainTextColor">{{
            dataset.data_source_name
          }}</span>
        </p>
        <p class="text is-caption is-txtMainTextColor" v-else>
          <span class="title is-txtMainTextColor">Geography</span>
        </p>
      </nav>
      <h1 class="title is-sectiontitle is-txtMainTextColor u-mt--4">
        {{ dataset.name }}
      </h1>
    </div>

    <div class="u-ml--auto grid-cell grid-cell--col3 grid-cell--col4--tablet">
      <div class="u-flex u-flex__justify--end">
        <Button
          v-if="getSubscriptionStatus === 'interested'"
          :url="getFormUrl()"
        >
          Iâ€™m interested
        </Button>
        <Button
          v-else-if="getSubscriptionStatus === 'free_subscription'"
          @click.native="showModal('subscribe')"
        >
          Subscribe for free
        </Button>
        <Button
          v-else-if="getSubscriptionStatus === 'request_subscription'"
          @click.native="showModal('request')"
        >
          Request subscription
        </Button>
        <div
          v-else-if="getSubscriptionStatus === 'active'"
          class="u-flex u-flex__direction--column u-flex__align--center"
        >
          <Button class="is-outline extra-border navy-blue noCursor">
            Subscribed
            <img class="u-ml--12" src="../../assets/check.svg" alt="check" />
          </Button>
          <span
            @click.native="showModal('unsubscribe')"
            class="text is-small is-txtSoftGrey u-mt--8 underline"
            >Unsubscribe</span
          >
        </div>
        <Button
          v-else-if="getSubscriptionStatus === 'requested'"
          class="is-outline extra-border navy-blue noCursor"
        >
          Requested
          <img class="u-ml--12" src="../../assets/check.svg" alt="check" />
        </Button>
      </div>
      <p
        v-if="subscriptionInfo.status !== 'active'"
        class="text is-small is-txtMainTextColor u-mt--16 right-align"
      >
        Any questions? <a href="/">Contact</a>
      </p>
    </div>

    <ModalSubscription
      @closeModal="hideModal()"
      :isOpen="modalOpen"
      :dataset="dataset"
      :mode="modalMode"
    ></ModalSubscription>
  </header>
</template>

<script>
import { mapState } from 'vuex';
import Button from '../Button';
import ModalSubscription from '../ModalSubscription';
import { formUrl } from '../../utils/form-url';

export default {
  name: 'DatasetHeader',
  data() {
    return {
      modalOpen: false,
      modalMode: null
    };
  },
  components: {
    Button,
    ModalSubscription
  },
  computed: {
    ...mapState({
      dataset: state => state.doCatalog.dataset,
      subscriptionInfo: state => state.doCatalog.subscriptionInfo
    }),
    isPublicWebsite() {
      return !(this.$store.state.user && this.$store.state.user.id);
    },
    isGeography() {
      return this.$route.params.type === 'geography';
    },
    getSubscriptionStatus() {
      const possibleLicenceStates = ['requested', 'active', 'expired'];
      if (
        !this.isPublicWebsite &&
        this.subscriptionInfo.status &&
        possibleLicenceStates.indexOf(this.subscriptionInfo.status) >= 0
      ) {
        return this.subscriptionInfo.status;
      }
      if (this.isPublicWebsite || !this.$store.state.user.is_enterprise) {
        return 'interested';
      } else if (
        this.$store.state.user.is_enterprise &&
        this.dataset.is_public_data !== undefined
      ) {
        return this.dataset.is_public_data
          ? 'free_subscription'
          : 'request_subscription';
      }
      return null;
    }
  },
  methods: {
    getFormUrl() {
      return formUrl(
        this.dataset.category_name,
        this.dataset.country_name,
        this.dataset.data_source_name
      );
    },
    showModal(mode) {
      this.modalMode = mode;
      this.modalOpen = true;
    },
    hideModal() {
      this.modalMode = null;
      this.modalOpen = false;
    },
    fetchSubscriptionInfo() {
      if (this.dataset && this.dataset.id && !this.isPublicWebsite) {
        this.$store.dispatch('doCatalog/fetchSubscriptionInfo', {
          id: this.dataset.id,
          type: this.isGeography ? 'geography' : 'dataset',
          api_key: this.$store.state.user.api_key
        });
      }
    }
  },
  watch: {
    dataset() {
      this.fetchSubscriptionInfo();
    }
  },
  destroyed() {
    this.$store.commit('doCatalog/resetSubscriptionInfo');
  }
};
</script>

<style lang="scss" scoped>
@import '../../styles/variables';
.right-align {
  text-align: right;
}

.u-ml--auto {
  margin-left: auto;
}

.underline {
  text-decoration: underline;
}
</style>
