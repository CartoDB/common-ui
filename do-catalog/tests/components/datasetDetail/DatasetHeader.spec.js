import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';
import DatasetHeader from '@/components/datasetDetail/DatasetHeader';
import dataset from '../../fixtures/dataset';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('DatasetHeader.vue', () => {
  const store = createStore();

  it('Renders view properly', () => {
    const datasetHeader = shallowMount(DatasetHeader, {
      localVue,
      store
    });
    expect(datasetHeader).toMatchSnapshot();
  });
});

function createStore(data = dataset.dataset_samples[0]) {
  const store = new Vuex.Store({
    state: {
      catalog: {
        dataset: data
      }
    }
  });
  return store;
}
