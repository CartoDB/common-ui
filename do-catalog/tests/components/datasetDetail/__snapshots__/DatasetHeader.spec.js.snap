// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetHeader.vue Renders view properly 1`] = `
<header class="grid u-flex__justify--center u-mb--36 u-mt--36">
  <div class="grid-cell grid-cell--col7">
    <nav class="breadcrumbs">
      <p class="text is-caption is-txtMainTextColor"><a class="title is-txtMainTextColor">Demographics</a> /
        <a class="text is-txtMainTextColor">Retail Centrality</a></p>
    </nav>
    <h2 class="title is-subheader is-txtMainTextColor u-mt--4">Retail Centrality at PC 5-digit (postal) level for USA</h2>
  </div>
  <div class="grid-cell--col1"></div>
  <div class="grid-cell--col2">
    <div class="u-flex u-flex__justify--end"><button class="button">Subscribe for Free</button></div>
    <p class="text is-small is-txtMainTextColor u-mt--16 right-align">Any questions? <a href="/">Contact</a></p>
  </div>
</header>
`;
